src/main/java/uk/gov/hmcts/reform/unspec/handler/RequestExtensionCallbackHandler.java

```diff
@@ -0,0 +1,101 @@
+package uk.gov.hmcts.reform.unspec.handler;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.SubmittedCallbackResponse;
+import uk.gov.hmcts.reform.unspec.callback.Callback;
+import uk.gov.hmcts.reform.unspec.callback.CallbackHandler;
+import uk.gov.hmcts.reform.unspec.callback.CallbackParams;
+import uk.gov.hmcts.reform.unspec.callback.CallbackType;
+import uk.gov.hmcts.reform.unspec.callback.CaseEvent;
+import uk.gov.hmcts.reform.unspec.enums.YesOrNo;
+import uk.gov.hmcts.reform.unspec.validation.RequestExtensionValidator;
+
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.String.format;
+import static java.time.LocalDate.now;
+import static uk.gov.hmcts.reform.unspec.callback.CaseEvent.REQUEST_EXTENSION;
+import static uk.gov.hmcts.reform.unspec.enums.YesOrNo.YES;
+import static uk.gov.hmcts.reform.unspec.helpers.DateFormatHelper.DATE;
+import static uk.gov.hmcts.reform.unspec.helpers.DateFormatHelper.formatLocalDate;
+
+@Service
+public class RequestExtensionCallbackHandler extends CallbackHandler {
+    private static final List<CaseEvent> EVENTS = Collections.singletonList(REQUEST_EXTENSION);
+    public static final String ALREADY_AGREED = "You told us you've already agreed this with the claimant's legal "
+        + "representative. We'll contact them and email you to confirm the deadline.</p>";
+    public static final String NOT_AGREED = "We'll email you to tell you if the claimant's legal representative "
+        + "accepts or rejects your request.</p>";
+
+    private final ObjectMapper mapper;
+    private final RequestExtensionValidator validator;
+
+    public RequestExtensionCallbackHandler(ObjectMapper mapper, RequestExtensionValidator validator) {
+        this.mapper = mapper;
+        this.validator = validator;
+    }
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return Map.of(
+            CallbackType.ABOUT_TO_START, this::aboutToStart,
+            CallbackType.MID, this::validateRequestedDeadline,
+            CallbackType.SUBMITTED, this::buildConfirmation
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    private CallbackResponse aboutToStart(CallbackParams callbackParams) {
+        CaseDetails caseDetails = callbackParams.getRequest().getCaseDetails();
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
```

Comment:
no need to pass data as nothing is modified

Reply:
is it not good practice to always add data? What do we think?

Reply:
removed